src/connector/https.rs

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::connector::http::{get_host, ConnectError, HttpConnection, HttpConnector};
+use crate::connector::{NetworkConnection, NetworkConnector};
+
+use hyper::client::connect::{Connected, Connection};
+use hyper::Uri;
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+use tokio::net::TcpStream;
+use tokio_native_tls::{TlsConnector, TlsStream};
+
+use std::error::Error as StdError;
+use std::future::Future;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// An HTTPS connector using native-tls
+pub struct HttpsConnector {
+    force_tls: bool,
+    tls: TlsConnector,
+}
+
+impl HttpsConnector {
+    pub fn new(tls: TlsConnector) -> Self {
+        HttpsConnector {
+            tls,
+            force_tls: false,
+        }
+    }
+
+    /// If called, the connector will ensure the URLs have the `https` scheme.
+    pub fn force_tls(mut self) -> Self {
+        self.force_tls = true;
```

Comment:
Having `force_tls` set to `true` by default would be safer, no?

Reply:
Sure